services/src/main/java/org/fao/geonet/config/DefaultPublicationConfig.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2001-2023 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+package org.fao.geonet.config;
+
+import org.fao.geonet.domain.ReservedGroup;
+import org.fao.geonet.domain.ReservedOperation;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Bean with the default publication configuration.
+ *
+ * The default publication options publishes the metadata to the ALL group and the INTRANET group.
+ */
+public class DefaultPublicationConfig implements IPublicationConfig {
```

Comment:
For discussion:

One limitation here, is that we can't add an option to publish in group "opendata" for example or "INSPIRE", right ? We are limited to internet/intranet/guest groups ? At least maybe not here in the config, but in the UI. And quite often this kind of privileges are assigned to push to a portal or to be harvested by third party apps.


Not sure if the approach taken by Vlaanderen adding custom workflow status (to publish internal, publish to all, delete, ...) is not more flexible. We made some improvements in https://github.com/geonetwork/core-geonetwork/pull/7011 to make the workflow more easy to extend. From the UI, it is easy to add new workflow status which can trigger same actions as in here. BTW, we have stuff to improve:
* no simple support to apply any status on a selection
* more UI rules to apply to check if status can be applied or not (here eg. show publishToIntranet (if not already published) - could be rules similar to the DOI task

Reply:
> One limitation here, is that we can't add an option to publish in group "opendata" for example or "INSPIRE", right ? We are limited to internet/intranet/guest groups ? At least maybe not here in the config, but in the UI. And quite often this kind of privileges are assigned to push to a portal or to be harvested by third party apps.

I can check to extend the configuration for any other group.

> Not sure if the approach taken by Vlaanderen adding custom workflow status (to publish internal, publish to all, delete, ...) is not more flexible. We made some improvements in https://github.com/geonetwork/core-geonetwork/pull/7011 to make the workflow more easy to extend. From the UI, it is easy to add new workflow status which can trigger same actions as in here.

Not everyone uses the metadata workflow.

> BTW, we have stuff to improve:
> - no simple support to apply any status on a selection
> - more UI rules to apply to check if status can be applied or not (here eg. show publishToIntranet (if not already published) - could be rules similar to the DOI task

Not sure what you mean with status, are you referring to the metadata status or you mean if the metadata is published or not?

Reply:
> I can check to extend the configuration for any other group.

Not needed now, but I've quite some examples of the privileges are used for this kind of changes and the point here is to avoid the "scary" sharing panel :) We can go with this for now, but I would maybe use status for that on the long run which are more flexible ... I'm new to the workflow usage for real - so I'm learning.

> Not everyone uses the metadata workflow.

We could have status actions which are not related to the workflow/draft stuff. A publishAction or publishToIntranetAction could be added in the database and will popup in the menu - like the DOI task.

> Not sure what you mean with status, are you referring to the metadata status or you mean if the metadata is published or not?

Sometimes you want to allow a status if something - eg you can only approve a record which is valid (so a check is done server side but a check can also be done client side to hide/disable the action). Currently this is not simple to plugin for each workflow status menu items.

Reply:
> We could have status actions which are not related to the workflow/draft stuff. A publishAction or publishToIntranetAction could be added in the database and will popup in the menu - like the DOI task.

Ok, but not sure if that will support publication to multiple groups. Can be these actions configured in the database?

For example, currently the publishAction publishes to `ALL` and `INTRANET` groups. We have customers where we require that the publish action publishes also to several groups.

I'm fine to change the current implementation to use that table as long as can be configured the actions.

> Sometimes you want to allow a status if something - eg you can only approve a record which is valid (so a check is done server side but a check can also be done client side to hide/disable the action). Currently this is not simple to plugin for each workflow status menu items.

Sure, that is not very well supported in the UI. For publication, if the workflow is enabled and the option to `Allow publication of invalid metadata` is disabled, the publish option is enabled / disabled based on the metadata validation status.

But there is room for improvement.

Reply:
> Ok, but not sure if that will support publication to multiple groups. Can be these actions configured in the database?

What has been done in Vlaanderen workflow is to declare new status and DefaultStatusAction is extended to check pre-conditions and apply the changes. So the privilege changes if any is done in the status action and can deal with many groups if needed - but need to change Java and not only configure a bean.

We can go with this first.